projects/tinkerbell/boots/patches/0001-Add-environment-variables-for.patch

```diff
@@ -0,0 +1,549 @@
+ 	if j.CanWorkflow() {
+ 		buildWorkerParams()
+ 		s.Args("docker_registry=" + dockerRegistry)
++		if len(registryUsername) > 0 {
++			s.Args("registry_username=" + registryUsername)
++			s.Args("registry_password=" + registryPassword)
++		}
++		if len(registryCertUrl) > 0 {
++			s.Args("registry_cert_url=" + registryCertUrl)
++		}
+ 		s.Args("grpc_authority=" + grpcAuthority)
+ 		s.Args("grpc_cert_url=" + grpcCertURL)
+ 		s.Args("instance_id=" + j.InstanceID())
+-		s.Args("registry_username=" + registryUsername)
+-		s.Args("registry_password=" + registryPassword)
+ 		s.Args("packet_base_url=" + workflowBaseURL())
+ 		s.Args("worker_id=" + j.HardwareID().String())
++		if len(tinkWorkerImage) > 0 {
++			s.Args("tink_worker_image=" + tinkWorkerImage)
++		}
+ 	}
+ 
+ 	s.Args("packet_bootdev_mac=${bootdevmac}")
+@@ -190,6 +198,9 @@ func isCustomOSIE(j job.Job) bool {
+ 
+ // osieBaseURL returns the value of Custom OSIE Service Version or just /current
+ func osieBaseURL(j job.Job) string {
++	if osiePathOverride != "" {
++		return osiePathOverride
++	}
+ 	if u := j.OSIEBaseURL(); u != "" {
+ 		return u
+ 	}
+diff --git a/installers/osie/mirror.go b/installers/osie/mirror.go
+index 796beba..672143b 100644
+--- a/installers/osie/mirror.go
++++ b/installers/osie/mirror.go
+@@ -14,11 +14,15 @@ const (
+ )
+ 
+ var (
+-	osieURL                            = mustBuildOSIEURL().String()
+-	mirrorBaseURL                      = conf.MirrorBaseUrl
+-	dockerRegistry                     string
+-	grpcAuthority, grpcCertURL         string
+-	registryUsername, registryPassword string
++	osieURL                    = mustBuildOSIEURL().String()
++	osiePathOverride           = getOSIEPathOverride()
++	mirrorBaseURL              = conf.MirrorBaseUrl
++	dockerRegistry             string
++	grpcAuthority, grpcCertURL string
++	registryUsername           string
++	registryPassword           string
++	registryCertUrl            string		
++	tinkWorkerImage            string
+ )
+ 
+ func buildOSIEURL() (*url.URL, error) {
+@@ -42,6 +46,22 @@ func buildOSIEURL() (*url.URL, error) {
+ 	return u, nil
+ }
+ 
++func getOSIEPathOverride() string {
++	base, err := url.Parse(conf.MirrorBaseUrl)
++	if err != nil {
++		panic(errors.Wrap(err, "parsing MirrorBaseUrl"))
++	}
++	if s, ok := os.LookupEnv("OSIE_PATH_OVERRIDE"); ok {
++		u, err := base.Parse(s)
++		if err != nil {
++			panic(errors.Wrapf(err, "invalid OSIE_PATH_OVERRIDE: %s", s))
++		}
++
++		return u.String() 
++	}
++	return ""
++}
++
+ func mustBuildOSIEURL() *url.URL {
+ 	u, err := buildOSIEURL()
+ 	if err != nil {
+@@ -55,8 +75,10 @@ func buildWorkerParams() {
+ 	dockerRegistry = getParam("DOCKER_REGISTRY")
+ 	grpcAuthority = getParam("TINKERBELL_GRPC_AUTHORITY")
+ 	grpcCertURL = getParam("TINKERBELL_CERT_URL")
+-	registryUsername = getParam("REGISTRY_USERNAME")
+-	registryPassword = getParam("REGISTRY_PASSWORD")
++	registryUsername = os.Getenv("REGISTRY_USERNAME")
```

Comment:
The `getParam()` method does call `os.Getenv()`
Was there a reason to not use it?
https://github.com/tinkerbell/boots/blob/v0.6.0/installers/osie/mirror.go#L63

Reply:
One of the purposes of this PR is to make registry username and password optional. getParam has a check to make sure they are not blank.